python_modules/libraries/dagster-managed-elements/dagster_managed_elements/cli.py

```diff
@@ -0,0 +1,89 @@
+import importlib.util
+import sys
+from types import ModuleType
+from typing import List
+
+import click
+from dagster_managed_elements.types import ManagedElementDiff, ManagedElementReconciler
+
+MODULE_NAME = "usercode"
+
+
+def load_module(file_path: str) -> ModuleType:
+    """
+    Imports a Python module from a file path.
+    https://docs.python.org/3/library/importlib.html#importing-a-source-file-directly
+    """
+    spec = importlib.util.spec_from_file_location(MODULE_NAME, file_path)
+    if not spec:
+        raise ValueError(f"Could not load module spec from {file_path}")
+    module = importlib.util.module_from_spec(spec)
+    sys.modules[MODULE_NAME] = module
+    loader = spec.loader
+    if not loader:
+        raise ValueError(f"Could not load loader from {file_path}")
+    loader.exec_module(module)
+
+    return module
+
+
+def get_reconcilable_objects(module: ModuleType) -> List[ManagedElementReconciler]:
+    """
+    Collect all ManagedElementReconciler-implementing objects in the root of the
+    module.
+    """
+    return [
+        getattr(module, obj)
+        for obj in dir(module)
+        if isinstance(getattr(module, obj), ManagedElementReconciler)
+    ]
```

Comment:
what led to this approach? Messing with imports can get spooky, might be nice to avoid if possible.

Reply:
marking resolved, continuing discussion in thread below